c7n/resources/elasticbeanstalk.py

```diff
@@ -55,8 +68,159 @@ class resource_type(object):
         filter_name = 'EnvironmentNames'
         filter_type = 'list'
 
+    filter_registry = env_filters
+    action_registry = env_actions
+    retry = staticmethod(get_retry(('ThrottlingException',)))
+    permissions = ('elasticbeanstalk:ListTagsForResource',)
+
+    def augment(self, envs):
+        filter(None, _eb_env_tags(
+            envs, self.session_factory, self.executor_factory, self.retry
+        ))
+        return envs
+
+
+def _eb_env_tags(envs, session_factory, executor_factory, retry):
+    """Augment ElasticBeanstalk Environments with their tags."""
+
+    def process_tags(eb_env):
+        client = local_session(session_factory).client('elasticbeanstalk')
+        try:
+            tag_list = retry(
+                client.list_tags_for_resource,
+                ResourceArn=eb_env['EnvironmentArn']
+            )['ResourceTags']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
```

Comment:
this should be != .. else where not logging the unknown error, ideally we should also log the eb_env thats caused the error as well.

Reply:
I thought this was weird, but I copied the pattern from [rdscluster](https://github.com/capitalone/cloud-custodian/blob/d7c44af8601239ab349fca3ef9aca41e7f25fb35/c7n/resources/rdscluster.py#L85) under the assumption that it was correct, and I just didn't understand why...

Updating.